File: Cap 04/4.6 - Operadores Logicos/4.6.4 - Operador Logico de Disjuncao Exclusiva/xortest.pas
{*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 4 - Topico 4.6.4
 * 
 * Programa ..: OPERACAO DE DISJUNCAO EXCLUSIVA ".XOU. / XOR"
 *}

program TestaLogicaXor;

var
  P1, P2: Integer;
  
begin
  writeln('TESTE LOGICO OPERADOR: XOR');
  writeln;
  
  writeln('Entre "1" se atleta pontuou na 1a. prova');
  writeln('Entre "1" se atleta pontuou na 2a. prova');
  writeln('Qualquer outro valor se nao pontuou nas provas');
  writeln;
  
  write('Prova 1: ');
  readln(P1);
  
  write('Prova 2: ');
  readln(P2);
  
  if (P1 = 1) xor (P2 = 1) then
  begin
    writeln('Atleta participa da terceira prova.');
  end
  else
  begin
    writeln('Atleta nao participa da terceira prova.');
    if (P1 = 1) and (P2 = 1) then
    begin
      writeln('Classificado para a final.');
    end
    else
    begin
      writeln('Desclassificado da competicao.');
    end;
  end;
  
end.

File: Cap 04/4.6 - Operadores Logicos/4.6.4 - Operador Logico de Disjuncao Exclusiva/xortest.vb
' Livro .....: Algorimos
'              Logica para Desenvolvimento de Programacao de 
'              Computadores
' Autores ...: Jose Augusto N. G. Manzano
'              Jayr Figueiredo de Oliveira
' Editora ...: Erica
' 
' Capitulo ..: 4 - Topico 4.6.4
' 
' Programa ..: OPERACAO DE DISJUNCAO EXCLUSIVA ".XOU. / XOR"

Module TesteLogico

    Sub Main()

        Dim P1 As Integer
        Dim P2 As Integer
        
        Console.WriteLine("TESTE LOGICO OPERADOR: XOR")
        Console.WriteLine()

        Console.WriteLine("Entre ""1"" se atleta pontuou na 1a. prova")
        Console.WriteLine("Entre ""1"" se atleta pontuou na 2a. prova")
        Console.WriteLine("Qualquer outro valor se nao pontuos nas provas")
        Console.WriteLine()

        Console.Write("Prova 1: ")
        P1 = Console.ReadLine()

        Console.Write("Prova 2: ")
        P2 = Console.ReadLine()

        If (P1 = 1) Xor (P2 = 1) Then
            Console.WriteLine("Atleta participa da terceira prova.")
        Else
            Console.WriteLine("Atleta nao participa da terceira prova.")
            If (P1 = 1) And (P2 = 1) Then
                Console.WriteLine("Classificado para a final.")
            Else
                Console.WriteLine("Desclassificado da competicao.")
            End If
        End If

    End Sub

End Module
